src/cirrus/lib2/eventdb.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import os
+from datetime import datetime
+from enum import Enum, unique
+from typing import Any, Dict, Optional
+
+import boto3
+from dateutil.parser import isoparse
+
+logger = logging.getLogger(__name__)
+
+
+@unique
+class StateEnum(Enum):
+    PROCESSING = "PROCESSING"
+    COMPLETED = "COMPLETED"
+    FAILED = "FAILED"
+    INVALID = "INVALID"
+    ABORTED = "ABORTED"
+
+
+class EventDB:
+    def __init__(
+        self,
+        event_db_and_table_names: Optional[str] = os.getenv(
+            "CIRRUS_EVENT_DB_AND_TABLE", None
+        ),
+    ):
+        self.tsw_client = boto3.client("timestream-write")
+        self.tsq_client = boto3.client("timestream-query")
+        db_and_table_names_array = event_db_and_table_names.split("|")
+        self.event_db_name = db_and_table_names_array[0]
+        self.event_table_name = db_and_table_names_array[1]
+
+    def write_timeseries_record(
+        self,
+        key: Dict[str, str],
+        state: StateEnum,
+        event_time: str,
+        last_update_ts_str: str,
+    ) -> None:
+        collections_workflow = key.get("collections_workflow")
+        itemids = key.get("itemids")
+
+        if not collections_workflow or not itemids:
+            logger.error(
+                f"Event could not be recorded, key {key} missing 'collections_workflow' or 'itemids'"
+            )
+            return
+
+        event_time_dt = datetime.fromisoformat(event_time)
+        event_time_ms = str(int(event_time_dt.timestamp() * 1000))
+        duration_ms = str(
+            int((isoparse(last_update_ts_str) - event_time_dt).microseconds / 1000)
+        )
+
+        record = {
+            "Dimensions": [
+                {"Name": "collections_workflow", "Value": collections_workflow},
```

Comment:
Could we split this to be `workflow` as the first dimension and `collections` as the second?

Reply:
yes